homeassistant/components/media_player/onkyo.py

```diff
@@ -61,20 +89,33 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     if CONF_HOST in config and host not in KNOWN_HOSTS:
         try:
+            receiver = eiscp.eISCP(host)
             hosts.append(OnkyoDevice(
-                eiscp.eISCP(host), config.get(CONF_SOURCES),
+                receiver,
+                config.get(CONF_SOURCES),
                 name=config.get(CONF_NAME),
                 max_volume=config.get(CONF_MAX_VOLUME),
             ))
             KNOWN_HOSTS.append(host)
 
-            # Add Zone2 if configured
-            if config.get(CONF_ZONE2):
+            zones = determine_zones(receiver)
+
+            # Add Zone2 if available
+            if zones["zone2"]:
                 _LOGGER.debug("Setting up zone 2")
-                hosts.append(OnkyoDeviceZone2(eiscp.eISCP(host),
-                                              config.get(CONF_SOURCES),
-                                              name=config.get(CONF_NAME) +
-                                              " Zone 2"))
+                hosts.append(OnkyoDeviceZone("2",
+                                             receiver,
+                                             config.get(CONF_SOURCES),
+                                             name=config.get(CONF_NAME) +
```

Comment:
- If you know that a key exists in a dict, use `config[CONF_NAME]`. 
- Use `"{} Zone 2".format(config[CONF_NAME])`

Reply:
Cheers, this was just copy pasta - I've updated 👍